src/main/java/net/logstash/logback/composite/loggingevent/StackHashJsonProvider.java

```diff
@@ -0,0 +1,97 @@
+ * limitations under the License.
+ */
+package net.logstash.logback.composite.loggingevent;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import ch.qos.logback.classic.spi.ThrowableProxy;
+import com.fasterxml.jackson.core.JsonGenerator;
+import net.logstash.logback.composite.AbstractFieldJsonProvider;
+import net.logstash.logback.composite.JsonWritingUtils;
+import net.logstash.logback.stacktrace.StackElementFilter;
+import net.logstash.logback.stacktrace.StackHasher;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * A JSON provider that adds a {@code stack_hash} Json field on a log with a stack trace
+ * <p>
+ * This hash is computed using {@link StackHasher}
+ * 
+ * @author Pierre Smeyers
+ */
+public class StackHashJsonProvider extends AbstractFieldJsonProvider<ILoggingEvent> {
+
+    public static final String FIELD_NAME = "stack_hash";
+
+    /**
+     * Patterns used to determine which stacktrace elements to exclude.
+     *
+     * The strings being matched against are in the form "fullyQualifiedClassName.methodName"
+     * (e.g. "java.lang.Object.toString").
+     *
+     * Note that these elements will only be excluded if and only if
+     * more than one consecutive line matches an exclusion pattern.
+     */
+    private List<Pattern> excludes = new ArrayList<Pattern>(5);
+
+    private StackHasher hasher = new StackHasher();
+
+    public StackHashJsonProvider() {
+        setFieldName(FIELD_NAME);
+    }
+
+    @Override
+    public void start() {
+        if(!excludes.isEmpty()) {
+            hasher = new StackHasher(StackElementFilter.byPattern(excludes));
+        }
+        super.start();
+    }
+
+    public void addExclude(String exclusionPattern) {
+        excludes.add(Pattern.compile(exclusionPattern));
+    }
+
+    public void setExcludes(List<String> exclusionPatterns) {
+        if (exclusionPatterns == null || exclusionPatterns.isEmpty()) {
+            this.excludes = new ArrayList<Pattern>(5);
+        } else {
+            this.excludes = new ArrayList<Pattern>(exclusionPatterns.size());
+            for (String pattern : exclusionPatterns) {
+                addExclude(pattern);
+            }
+        }
+    }
+
+    public List<String> getExcludes() {
+        List<String> exclusionPatterns = new ArrayList<String>(excludes.size());
+        for (Pattern pattern : excludes) {
+            exclusionPatterns.add(pattern.pattern());
+        }
+        return exclusionPatterns;
+    }
+
+    @Override
+    public void writeTo(JsonGenerator generator, ILoggingEvent event) throws IOException {
+        IThrowableProxy throwableProxy = event.getThrowableProxy();
+        if (throwableProxy != null && throwableProxy instanceof  ThrowableProxy) {
+        	String hash = hasher.hexHash(((ThrowableProxy)event.getThrowableProxy()).getThrowable());
```

Comment:
Yes. That's rule 2 in my comment:

2. an error hash shall compute against it's parent cause (recurses)